crates/uv-distribution/src/metadata/lowering.rs

```diff
@@ -140,6 +141,7 @@ impl LoweredRequirement {
             if matches!(lower_bound, LowerBound::Warn) {
                 // Support recursive editable inclusions.
                 if has_sources
+                    && !constrained_packages.contains(&requirement.name)
                     && requirement.version_or_url.is_none()
```

Comment:
So, should we remove this? I think it's redundant now? (Also, isn't this wrong? It seems to only look at `None`, and not the lower bound?)